lib/Mojo/Exception.pm

```diff
@@ -26,15 +26,16 @@ sub check {
 
   my ($default, $handler);
   my $is_obj = blessed $err;
+  my $can_re = !$is_obj || $is_obj && overload::Method($err, '""');
 CHECK: for (my $i = 0; $i < @spec; $i += 2) {
     my ($checks, $cb) = @spec[$i, $i + 1];
 
     ($default = $cb) and next if $checks eq 'default';
 
     for my $c (ref $checks eq 'ARRAY' ? @$checks : $checks) {
       my $is_re = ref $c eq 'Regexp';
```

Comment:
The proper way to check for regexes, that will work on all perl versions, is `re->can('is_regexp') ? re::is_regexp($_[0]) : ref($_[0]) eq 'Regexp'`

Reply:
We don't have to support all Perl versions.

Reply:
re::is_regexp became available in perl 5.9.5, so that would be preferrable to checking ref eq 'Regexp' in all cases then.